### ISSUE TITLE
Migrate away from react-bootstrap modals

### ISSUE THREAD

**Issue by mikhael28** [mikhael28/paretOS#87]
At some point, we want to get away from the modals in react-bootstrap 0.32 - they use outdated componentWillReceive props. Perhaps we can replace them with the `Modal` component inside the `components` folder. There are a lot of these, 6 or 7.

**Comment by akashshyamdev** [comment]
@mikhael28 Yes, great point. One thing I've noticed across the codebase is that you are using both material UI and react bootstrap. Why is that? Won't that unnecessarily increase bundle size? I think we should move forward with either MUI or react-bootstrap

**Comment by mikhael28** [comment]
That is a false dichotomy - if you open up the source-bundle explorer, you will notice that each of those UI frameworks are using only a very small number of kilobytes. This is because the code is set up to 'tree shake' very efficiently. If you will notice, when we are importing bootstrap components, we are importing them like `react-bootstrap/lib/Button' - by importing them from the '/lib/component' we are ONLY importing that component. If we were to lazily import everything like `import { Button, Form } from 'react-bootstrap'` then that will automatically import the ENTIRE library, which will drastically increase the size. The same goes for Material UI. So, we can import from multiple libraries - there is nothing wrong with that.

**Comment by mikhael28** [comment]
Now, that being said, I would like to move away from the modals and in the future - I would like to replace the react-bootstrap forms with 'react-hook-form', as that library is much more performant (fewer DOM re-renders) and has built in support for form validation, which we need to add over time.

**Comment by AleksandrKurilov** [comment]
Hi, Can I try to do it? Can you assign me to this one?

**Comment by mikhael28** [comment]
@AleksandrKurilov yes, I will assign it to you. Any questions?

**Comment by mikhael28** [comment]
Take a look at the `components/Modal.js` component and use that as a way of migrating modals. Would you also convert `Modal.js` to TypeScript?

**Comment by AleksandrKurilov** [comment]
Yes, I would. Thank you.

**Comment by mikhael28** [comment]
Great - let me know if you have any questions! Looking forward to seeing a PR - you also don't have to replace ALL of the modals at once, you can start with one - then send a PR, so we can talk about the implementation/the visual UI difference, and then we can merge that one in, and you can work on the others. Take it a step at a time/

**Comment by jayeclark** [comment]
I've added this to the Bootstrap → Material UI migration milestone! It looks like there are only 3 components left that are still using Bootstrap modals: 
- [ ] arena/ArenaProofModal (used in Sprint)
- [ ] learn/ApproveExperienceModal (used in ExperienceModule)
- [ ] learn/NewSubmitProofModal (used in ExperienceModule)

**Comment by jayeclark** [comment]
@AleksandrKurilov Were you still interested in working on one of these three? (Or, you're welcome to tackle all of them!)

**Comment by AleksandrKurilov** [comment]
@jayeclark Hello, I'm going to done all of them soon. Sorry I was a bit busy but now I'm ready done them. I will do them gradually as @mikhael28 sayid.

**Comment by AleksandrKurilov** [comment]
@mikhael28 @jayeclark , I have a question. Did I understand that right - I need to change ArenaProofModal, ApproveExperienceModal and NewSubmitProofModal so that they whether just use MUI or they use component/Modal.tsx ?

**Comment by jayeclark** [comment]
@AleksandrKurilov Thanks for the contribution!!

I'll let @mikhael28 weigh in but I think in the 8 months or so since this issue was opened, the repository has shifted toward just using the MUI Dialog modal instead of the `components/Modal.js`. 

Currently `components/Modal.js` is only being used as the modal component in a single file (TermsOfService), react-bootstrap Modal is still being used in 3, and the MUI Dialog component is being used as the modal in 6 files (so it seems like the winner!).

Refactoring the existing `components/Modal.js` to TypeScript is great - everything will be TypeScript eventually! I have to step away from my PC but will do my best to review your PR tonight!

For the three remaining files listed above, I'm pretty sure you're on the right path -- just use MUI. (Don't bother with components/Modal.tsx)

**Comment by mikhael28** [comment]
Yes, the Material UI modal has proved to be the 'fittest' of the Modal component. @AleksandrKurilov the best scenario would be to go ahead and migrate the TermsOfService modal to the Material UI Dialog? Then, you can delete the Modal.js/tsx?

**Comment by mikhael28** [comment]
LMAO it's been 8 months since I opened this ticket. Hilarious. Father Time waits for no one.

**Comment by jayeclark** [comment]
I’m excited to see some of these old ones finally getting closed out!

**Comment by mikhael28** [comment]
@AleksandrKurilov I wrote a separate note in the PR that you submitted - go ahead and resolve the merge conflict, and I can merge that code in. I don't mind keeping Modal.tsx in the codebase for the time being - I'm curious how many kb it is compared to the MUI Dialog.

**Comment by AleksandrKurilov** [comment]
I've made new PR with ArenaProofModal, but I didn't migrate it on TS. I will move it on TS in next PR.

**Comment by AleksandrKurilov** [comment]
The last one

- [x] arena/ArenaProofModal (used in Sprint)
- [x]  learn/ApproveExperienceModal (used in ExperienceModule)
- [x]  learn/NewSubmitProofModal (used in ExperienceModule)

Then I'm going to migrate these modals to TS
